zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/ExecRemoteInterpreterProcess.java

```diff
@@ -100,12 +100,23 @@ public void start(String userName) throws IOException {
       Matcher m = YARN_APP_PATTER.matcher(launchOutput);
       if (m.find()) {
         String appId = m.group(1);
-        LOGGER.info("Detected yarn app: {}, add it to YarnAppMonitor", appId);
-        YarnAppMonitor.get().addYarnApp(ConverterUtils.toApplicationId(appId), this);
+        if (isHadoopClientAvailable()) {
+          LOGGER.info("Detected yarn app: {}, add it to YarnAppMonitor", appId);
+          YarnAppMonitor.get().addYarnApp(ConverterUtils.toApplicationId(appId), this);
+        }
```

Comment:
No pattern matching is needed if hadoop client is not available.
It would be good to move this condition to line 98.